app/tools/google_sheets_manager.py

```
@@ -0,0 +1,68 @@
+    def __init__(self, spreadsheet_name=None):
+        """
+        Initializes the GoogleSheetsManager with service account credentials.
+        """
+        if not Config.SERVICE_ACCOUNT:
+            raise ValueError("Service account path is not set in the configuration.")
+        
+        self.client = gspread.service_account(filename=Config.SERVICE_ACCOUNT)
+        try:
+            self.spreadsheet = self.client.open_by_url(spreadsheet_name)
+        except Exception as e:
+            raise ValueError(f"Failed to open spreadsheet '{spreadsheet_name}': {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)